### ISSUE TITLE
Getting error while getting messages from DavMail

### ISSUE THREAD

**Issue by diogofgm** [jorritfolmer/TA-dmarc#31]
Splunk Version 8.1.1
TA-dmarc Version 3.2.4

I am connecting to DavMail via IMAP and I'm trying to get the DMARC attachments from a mailbox but I'm are getting decoding errors:

```
2021-01-27 16:25:25,992 ERROR pid=121493 tid=MainThread file=base_modinput.py:log_error:309 | Get error when collecting events.
Traceback (most recent call last):
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/aob_py3/modinput_wrapper/base_modinput.py", line 128, in stream_events
    self.collect_events(ew)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/dmarc_imap.py", line 92, in collect_events
    input_module.collect_events(self, ew)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/input_module_dmarc_imap.py", line 50, in collect_events
    filelist = i2d.process_incoming()
  File "/opt/splunk/etc/apps/TA-dmarc/bin/dmarc/imap2dir.py", line 305, in process_incoming
    filelist = self.save_reports_from_message_bodies(response)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/dmarc/imap2dir.py", line 194, in save_reports_from_message_bodies
    msg = email.message_from_string(data[b'RFC822'].decode("utf-8", "replace"))
KeyError: b'RFC822'
```

I enabled debug logs an I'm able to see the messages string starting like this:

```
2021-01-27 16:25:25,991 DEBUG pid=121493 tid=MainThread file=imaplib.py:_untagged_response:1207 | untagged_responses[FETCH] => [(b'6 (UID 1184 BODY[] {4250}', b'Received: from mx1.xxx.xxx (xxx.xxx.xxx.xxx)
(... it keeps going with all message details ...)
```

What am I missing here?

**Comment by jorritfolmer** [comment]
I think there are items in that inbox that aren't emails, and I haven't programmed that part of the code defensively enough it seems to handle that properly.

I'll see what I can do. Don't expect a fix in a matter of days but in weeks. Always happy to accept PRs btw.

**Comment by jorritfolmer** [comment]
I've fixed the issue in the most recent commit to master. Can you verify if the current version fixes your issue? I'll be releasing v3.2.5 later because there are currently some issues on Splunkbase.
